db/c.cc

```diff
@@ -2641,6 +2651,17 @@ rocksdb_iterator_t* rocksdb_writebatch_wi_create_iterator_with_base_cf(
   return result;
 }
 
+rocksdb_iterator_t* rocksdb_writebatch_wi_create_iterator_with_base_cf_readopts(
+    rocksdb_writebatch_wi_t* wbwi, rocksdb_iterator_t* base_iterator,
+    rocksdb_column_family_handle_t* column_family,
+    const rocksdb_readoptions_t* options) {
+  rocksdb_iterator_t* result = new rocksdb_iterator_t;
+  result->rep = wbwi->rep->NewIteratorWithBase(
+      column_family->rep, base_iterator->rep, &options->rep);
+  delete base_iterator;
```

Comment:
Same question about potential double free as above.

Reply:
I think this should be ok since deleting `rocksdb_iterator_t` will only delete the struct but not the pointed to iterator (there is no destructor defined which frees the pointed to iterator when the outer struct gets deleted).